homeassistant/components/switch/telnet.py

```diff
@@ -0,0 +1,154 @@
+
+from homeassistant.components.switch import (SwitchDevice, PLATFORM_SCHEMA,
+                                             ENTITY_ID_FORMAT)
+from homeassistant.const import (
+    CONF_RESOURCE, CONF_FRIENDLY_NAME, CONF_OPTIMISTIC, CONF_SWITCHES,
+    CONF_VALUE_TEMPLATE, CONF_COMMAND_OFF, CONF_COMMAND_ON, CONF_COMMAND_STATE,
+    CONF_PORT)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PORT = 23
+
+SWITCH_SCHEMA = vol.Schema({
+    vol.Required(CONF_RESOURCE): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Required(CONF_COMMAND_OFF): cv.string,
+    vol.Required(CONF_COMMAND_ON): cv.string,
+    vol.Optional(CONF_COMMAND_STATE): cv.string,
+    vol.Optional(CONF_FRIENDLY_NAME): cv.string,
+    vol.Optional(CONF_OPTIMISTIC): cv.boolean,
+    vol.Required(CONF_VALUE_TEMPLATE): cv.template,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SWITCHES): vol.Schema({cv.slug: SWITCH_SCHEMA}),
+})
+
+SCAN_INTERVAL = timedelta(seconds=10)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Find and return switches controlled by shell commands."""
+    devices = config.get(CONF_SWITCHES, {})
+    switches = []
+
+    for object_id, device_config in devices.items():
+        value_template = device_config.get(CONF_VALUE_TEMPLATE)
+
+        if value_template is not None:
+            value_template.hass = hass
+
+        switches.append(
+            TelnetSwitch(
+                hass,
+                object_id,
+                device_config.get(CONF_RESOURCE),
+                device_config.get(CONF_PORT),
+                device_config.get(CONF_FRIENDLY_NAME, object_id),
+                device_config.get(CONF_COMMAND_ON),
+                device_config.get(CONF_COMMAND_OFF),
+                device_config.get(CONF_COMMAND_STATE),
+                device_config.get(CONF_OPTIMISTIC),
+                value_template
+            )
+        )
+
+    if not switches:
+        _LOGGER.error("No switches added")
+        return False
+
+    add_devices(switches)
+
+
+class TelnetSwitch(SwitchDevice):
+    """Representation of a switch that can be toggled using telnet commands."""
+
+    def __init__(self, hass, object_id, resource, port, friendly_name,
+                 command_on, command_off, command_state, optimistic,
+                 value_template):
+        """Initialize the switch."""
+        self._hass = hass
+        self.entity_id = ENTITY_ID_FORMAT.format(object_id)
+        self._resource = resource
+        self._port = port
+        self._name = friendly_name
+        self._state = False
+        self._command_on = command_on
+        self._command_off = command_off
+        self._command_state = command_state
+        self._optimistic = optimistic
+        self._value_template = value_template
+
+    def __telnet_command(self, command):
```

Comment:
Small nitpick, but single underscore prefix is preferred for normal internal methods.